Print warning on .hsc files.

pparkkin opened the pull request:
Print out a warning if user passes in a `.hsc` file to the compiler. See issue https://github.com/typelead/eta/issues/97.

Should I add tests?

## Before

A `.hsc` file only.

```
€ eta Main.hsc
eta: panic! (the 'impossible' happened)
  (Eta version 0.0.5):
	Parsing of archive structure failed:
Cannot locate end of central directory
in "/home/vagrant/src/eta-paavo/hsc-test/Main.hsc"

Please report this as a Eta bug: http://github.org/typelead/eta/issues
```

Both `.hsc` and `.hs` files.

```
€ eta Main.hsc Main.hs
[1 of 1] Compiling Main             ( Main.hs, Main.jar )
Linking RunMain.jar ...
eta: panic! (the 'impossible' happened)
  (Eta version 0.0.5):
	Parsing of archive structure failed:
Cannot locate end of central directory
in "/home/vagrant/src/eta-paavo/hsc-test/Main.hsc"

Please report this as a Eta bug: http://github.org/typelead/eta/issues
```

## After

A `.hsc` file only.

```
€ eta Main.hsc
WARN: File Main.hsc of unsupported type (.hsc)
eta: panic! (the 'impossible' happened)
  (Eta version 0.0.5):
	Parsing of archive structure failed:
Cannot locate end of central directory
in "/home/vagrant/src/eta-paavo/hsc-test/Main.hsc"

Please report this as a Eta bug: http://github.org/typelead/eta/issues
```

Both `.hsc` and `.hs` files.

```
€ eta Main.hsc Main.hs
[1 of 1] Compiling Main             ( Main.hs, Main.jar )
Linking RunMain.jar ...
WARN: File Main.hsc of unsupported type (.hsc)
eta: panic! (the 'impossible' happened)
  (Eta version 0.0.5):
	Parsing of archive structure failed:
Cannot locate end of central directory
in "/home/vagrant/src/eta-paavo/hsc-test/Main.hsc"

Please report this as a Eta bug: http://github.org/typelead/eta/issues
```

rahulmutt commented on compiler/ETA/Main/DriverPipeline.hs:
```diff
@@ -1305,6 +1309,7 @@ linkGeneric dflags oFiles depPackages = do
             -- TODO: Verify that the right version eta was used
             --       in the Manifests of the jars being compiled
           else return []
+    mapM_ warnHscFile $ filter (".hsc" `isSuffixOf`) (jarInputs dflags)
```
Why do you check `jarInputs` as well for `.hsc` file extension?

pparkkin replied:
If I pass in a `.hs` file and a `.hsc` file, the `.hsc` file ends up in `jarInputs`. If I only pass in a `.hsc` file, it will be in `oFiles`.

rahulmutt commented on compiler/ETA/Main/DriverPipeline.hs:
```diff
@@ -1293,6 +1296,7 @@ linkGeneric dflags oFiles depPackages = do
     --        (text $ "    Call hsInit() from your main() method to set"
     --          ++ " these options."))
     -- TODO: Use conduits to combine the jars
+    mapM_ warnHscFile $ filter (".hsc" `isSuffixOf`) oFiles
```
This should be extended to fail (via throwing an exception) or use if-else and make sure nothing else happens if an `.hsc` file is detected.

pparkkin commented:
- Moved handling for `.hsc` files to `eta/Main.hs` to before anything gets compiled.
- Changed warning to error, and will exit without doing anything if any `.hsc` files are found.

rahulmutt commented:
This is looking good. Will merge once I update the build scripts and it passed at least Travis.
